### DISCUSSION TITLE
Appropriate Solvers for General Equation of State

### DISCUSSION THREAD

**Discussion by hwhitehead** [PrincetonUniversity/athena#474] [Q&A] [answered]
I am currently attempting to write a custom equation of state to allow for contributions from radiation pressure. Once this is complete the plan is to include appropriate heating and cooling effects. I am not including magnetic fields. 

I am trying to work out what solver would be most appropriate for this task. From the code [paper](https://iopscience.iop.org/article/10.3847/1538-4365/ab929b/pdf), section 3.2.2 suggests that HLLE, HLLC and HLLD have been made compatible with a general equation of state. However, the documentation in [hllc.cpp](https://github.com/PrincetonUniversity/athena/blob/master/src/hydro/rsolvers/hydro/hllc.cpp) states that it is only appropriate for adiabatic hydrodynamics. Further more, [configure.py](https://github.com/PrincetonUniversity/athena/blob/master/configure.py) throws an error when attempting to use a general EoS with anything other than HLLC or HLLD.

Is this documentaiton/error catching up to date? I would greatly appreciate any advice as to how best to robustly implement a gas-radiation mixture in my simulation.

**Comment by msbc** [comment, marked as answer]
HLLC or HLLD should be used with the general equation of state.  Both of these R-solvers support the general EOS. We opted to disable HLLE for a general EOS as it failed some specific Riemann tests.

**Reply by hwhitehead** [reply]
Thanks Matt, appreciate the speedy response.
